src/main/java/core/Engine.java

```diff
@@ -3,14 +3,15 @@
 import io.Input;
 import utils.TimeUtils;
 
+import java.util.ArrayList;
+
 public class Engine{
 
     public Window window;
-    public Entity entity;
+    public ArrayList<Entity> entities = new ArrayList<Entity>();
```

Comment:
```suggestion
    public List<Entity> entities = new ArrayList<Entity>();
```
Any specific reason to use `ArrayList`?
Use abstract interfaces if concrete implementation is not required.